### PR TITLE
default indicators can now accept custom text

### PR DESCRIPTION
**Pull request by abubakarmvp**
for translation purpose i updated the default indicator pages and also export them to be available in the app

### CODE REVIEW

**Comment by clragon** [comment]
could you update your PR to only include changes to the files necessary?

**Comment by abubakarmvp** [comment]
hi, 
i updated the PR

File: example/ios/Podfile.lock [hunk 1]
```diff
@@ -35,4 +35,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: ef19549a9bc3046e7bb7d2fab4d021637c0c58a3
 
-COCOAPODS: 1.11.3
```
Comment (clragon) [inline example/ios/Podfile.lock:38]: please remove changes to other files

File: example/lib/samples/beer_list_view.dart [hunk 2]
```diff
@@ -48,6 +48,9 @@ class _BeerListViewState extends State<BeerListView> {
           pagingController: _pagingController,
           builderDelegate: PagedChildBuilderDelegate<BeerSummary>(
             animateTransitions: true,
+            firstPageErrorIndicatorBuilder: (_) => FirstPageErrorIndicator(
+              onTryAgain: () => _pagingController.refresh(),
+            ),
```
Comment (clragon) [inline example/lib/samples/beer_list_view.dart:53]: is there a reason you have added this? 
this should be the default already, right?

File: lib/src/widgets/helpers/default_status_indicators/first_page_exception_indicator.dart [hunk 3]
```diff
@@ -4,15 +4,16 @@ import 'package:flutter/material.dart';
 class FirstPageExceptionIndicator extends StatelessWidget {
   const FirstPageExceptionIndicator({
     required this.title,
-    this.message,
+    required this.message,
```
Comment (clragon) [inline lib/src/widgets/helpers/default_status_indicators/first_page_exception_indicator.dart:7]: is there a reason you have nmade this required?

File: lib/src/widgets/helpers/default_status_indicators/first_page_exception_indicator.dart [hunk 4]
```diff
@@ -4,15 +4,16 @@ import 'package:flutter/material.dart';
...
     this.onTryAgain,
+    this.tryAgainBtnTxt,
```
Comment (clragon) [inline lib/src/widgets/helpers/default_status_indicators/first_page_exception_indicator.dart:9]: could this maybe named in a way that fits the other parameter names?
e.g. `tryAgainText`

File: lib/src/widgets/helpers/default_status_indicators/first_page_progress_indicator.dart [hunk 5]
```diff
@@ -1,11 +1,18 @@
 import 'package:flutter/material.dart';
 
 class FirstPageProgressIndicator extends StatelessWidget {
+  const FirstPageProgressIndicator({
+    Key? key,
+    this.adaptive = false,
+  }) : super(key: key);
+  final bool adaptive;
```
Comment (clragon) [inline lib/src/widgets/helpers/default_status_indicators/first_page_progress_indicator.dart:8]: could you add some spaces around this variable?

File: lib/src/widgets/helpers/default_status_indicators/new_page_error_indicator.dart [hunk 6]
```diff
@@ -5,24 +5,26 @@ class NewPageErrorIndicator extends StatelessWidget {
   const NewPageErrorIndicator({
     Key? key,
     this.onTap,
+    this.errorMsg = 'Something went wrong. Tap to try again.',
```
Comment (clragon) [inline lib/src/widgets/helpers/default_status_indicators/new_page_error_indicator.dart:8]: could we use full names for variables?
e.g. `errorMessage`?

File: lib/src/widgets/helpers/default_status_indicators/new_page_error_indicator.dart [hunk 7]
```diff
@@ -5,24 +5,26 @@ class NewPageErrorIndicator extends StatelessWidget {
...
   }) : super(key: key);
+  final String errorMsg;
```
Comment (clragon) [inline lib/src/widgets/helpers/default_status_indicators/new_page_error_indicator.dart:10]: could you add a space here?

File: lib/src/widgets/helpers/default_status_indicators/new_page_progress_indicator.dart [hunk 8]
```diff
@@ -4,10 +4,14 @@ import 'package:infinite_scroll_pagination/src/widgets/helpers/default_status_in
 class NewPageProgressIndicator extends StatelessWidget {
   const NewPageProgressIndicator({
     Key? key,
+    this.adaptive = false,
   }) : super(key: key);
+  final bool adaptive;
```
Comment (clragon) [inline lib/src/widgets/helpers/default_status_indicators/new_page_progress_indicator.dart:9]: could you add a space above here?

File: lib/src/widgets/helpers/default_status_indicators/no_items_found_indicator.dart [hunk 9]
```diff
@@ -3,9 +3,17 @@ import 'package:flutter/material.dart';
 import 'package:infinite_scroll_pagination/src/widgets/helpers/default_status_indicators/first_page_exception_indicator.dart';
 
 class NoItemsFoundIndicator extends StatelessWidget {
+  const NoItemsFoundIndicator({
+    Key? key,
+    this.title = 'No items found',
+    this.message = 'The list is currently empty.',
+  }) : super(key: key);
+
+  final String title;
```
Comment (clragon) [inline lib/src/widgets/helpers/default_status_indicators/no_items_found_indicator.dart:12]: could you add some spaces here?

**Review by clragon — CHANGES_REQUESTED** [review]

**Comment by abubakarmvp** [comment]
hi,
i updated the PR. can please check it.

**Comment by clragon** [comment]
This PR requires exporting the default page indicators.
I believe that might be undesirable, as indicated in #36.

#216 might be a better way of solving this.

**Pull request closed without merging**